src/xhr.js

```diff
@@ -1,31 +1,80 @@
 import { fetch } from './fetch.js';
 import { Evented } from './evented.js';
 
+// Extends from Evented to fire events
 export class XMLHttpRequest extends Evented {
-  open (method, url) {
+  constructor() {
+    super();
+    this.headers = {};
+    this.readyState = 1;
+    this.withCredentials = true;
+  }
+
+  // Stores method and url
+  open(method, url) {
     this.method = method;
     this.url = url;
   }
-  send () {
-    // print('XHR', this.method, this.url);
-    fetch(this.url, { method: this.method })
+
+  /**
+   * Resets status and values
+   * Needed by some implementations
+   */
+  abort() {
+    this.responseText = '';
+    this.response = null;
+    this.readyState = 0;
+    this.status = 0;
+  }
+
+  // Append header to headers object
+  setRequestHeader(...args) {
```

Comment:
why use `...args` and not `name, value` or something more descriptive?

Reply:
Actually no reason, I started it like that to test it but I'll make it more descriptive 👍